### PR TITLE
adds support for external link and image for page link page and adds migration

### LINKED ISSUE #662: "Link" Page Type
Link to internal Pages or external URLs that is listed in Sections along side Articles.

Not sure why this doesn't exist yet in Wagtail, potentially there are good reasons.

### LINKED ISSUE #894: During migration, "Image" designated in Footers should be in the place of the icon rather than the lead image
In mz.goodinternet.org:

image

In mz.iogt.site:

image

### LINKED ISSUE #895: Allow images as well as SVGs for icons
This is important because sometimes we need to respect brand guidelines. For example this is relevant for Facebook and U-Report brands on mz.goodinternet.org

image

These images should not change as theme settings change, eg color of text. To have an image that changes with text color, convert to SVG.

### PR DESCRIPTION
**Pull request by sheralim012**
fixes #662
fixes #895
fixes #894 

- adds raster-icon(PNG/JPEG) support in section, articles, footers, offline app page, page link page, flat menu item, poll, survey, quiz
  - SVG icon will take precedence over the raster icon
  - in case of page link page and flat menu item, if icon/raster-icon isn't given then linked page icon/raster-icon will be used
- v1 footer image will  be migrated as raster-icon in footer
- adds support for external link in page link page

### CODE REVIEW

File: home/models.py [hunk 1]
```diff
@@ -535,11 +542,14 @@ class PageLinkPage(Page, TitleIconMixin):
         on_delete=models.SET_NULL,
     )
 
-    page = models.ForeignKey(Page, related_name='page_link_pages', on_delete=models.PROTECT)
+    page = models.ForeignKey(Page, null=True, blank=True, related_name='page_link_pages', on_delete=models.PROTECT)
+    external_link = models.URLField(null=True, blank=True)
 
     content_panels = Page.content_panels + [
+        ImageChooserPanel('image'),
```
Comment (geoo89) [inline home/models.py:549]: Just to clarify, the idea is that in general, icons can be both SVG or PNG (for footers (which are now articles in V2), PageLinkPages, and in principle also sections, etc), and they would also be displayed like icons, just without the recoloring.

Comment (geoo89) [reply home/models.py:549]: Not sure if it's easier to implement this via two fields (e.g. icon and icon_image, where one of them takes priority) or one (which could be either format), or leave it for later and stick with the warning for now.

**Comment by geoo89** [comment]
Featured content in sections uses icons. When loading a section with featured content, it crashes:

```
Environment:


Request Method: GET
Request URL: http://localhost:8000/en/sections/emergency-information/

Django Version: 3.1.13
Python Version: 3.8.0
Installed Applications:
['home',
 'search',
 'iogt_users',
 'comments',
 'iogt_content_migration',
 'questionnaires',
 'messaging',
 'django.contrib.humanize',
 'wagtail_localize',
 'wagtail_localize.locales',
 'wagtail.contrib.forms',
 'wagtail.contrib.redirects',
 'wagtail.embeds',
 'wagtail.sites',
 'wagtail.users',
 'wagtail.snippets',
 'wagtail.documents',
 'wagtail.images',
 'wagtail.search',
 'wagtail.admin',
 'wagtail.core',
 'wagtail.contrib.modeladmin',
 'wagtailmenus',
 'wagtailmedia',
 'wagtailmarkdown',
 'wagtail_transfer',
 'wagtailsvg',
 'wagtail.contrib.settings',
 'django_comments_xtd',
 'django_comments',
 'modelcluster',
 'taggit',
 'allauth',
 'allauth.account',
 'allauth.socialaccount',
 'rest_framework',
 'sass_processor',
 'translation_manager',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.sites',
 'django_extensions']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'whitenoise.middleware.WhiteNoiseMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.locale.LocaleMiddleware',
 'wagtail.contrib.redirects.middleware.RedirectMiddleware',
 'iogt_users.middlewares.RegistrationSurveyRedirectMiddleware',
 'external_links.middleware.RewriteExternalLinksMiddleware',
 'iogt.middleware.CacheControlMiddleware']


Template error:
In template /data/idems/iogt/iogt/home/templates/home/tags/featured_content_card.html, error at line 5
   2
   1 : {% load image_tags %}
   2 : {% load static wagtailcore_tags wagtailimages_tags %}
   3 : 
   4 : {% if page.icon %}
   5 :      {% render_png_from_svg page.get_icon_url width=16 height=16 %} 
   6 : {% endif %}
   7 : <a href="{% pageurl page %}">{{ page.title }}</a>
   8 : {% include 'home/tags/angle_bracket.html' %}

Traceback (most recent call last):
  File "/data/idems/iogt/iogt/home/models.py", line 1130, in get_png_image
    obj = cls.objects.get(svg_path=svg_path, fill_color=fill_color, stroke_color=stroke_color)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/db/models/query.py", line 429, in get
    raise self.model.DoesNotExist(

During handling of the above exception (SVGToPNGMap matching query does not exist.), another exception occurred:
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/core/handlers/base.py", line 204, in _get_response
    response = response.render()
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/response.py", line 105, in render
    self.content = self.rendered_content
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/response.py", line 83, in rendered_content
    return template.render(context, self._request)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 170, in render
    return self._render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/loader_tags.py", line 150, in render
    return compiled_parent._render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
    return nodelist.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/loader_tags.py", line 192, in render
    return template.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 172, in render
    return self._render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/defaulttags.py", line 211, in render
    nodelist.append(node.render_annotated(context))
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/loader_tags.py", line 192, in render
    return template.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 172, in render
    return self._render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
    return nodelist.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/template/library.py", line 214, in render
    _dict = self.func(*resolved_args, **resolved_kwargs)
  File "/data/idems/iogt/iogt/home/templatetags/image_tags.py", line 13, in render_png_from_svg
    'image_url': SVGToPNGMap.get_png_image(absolute_path, fill_color=fill_color, stroke_color=stroke_color).url,
  File "/data/idems/iogt/iogt/home/models.py", line 1132, in get_png_image
    png_image = convert_svg_to_png_bytes(svg_path, fill_color=fill_color, stroke_color=stroke_color, width=32)
  File "/data/idems/iogt/iogt/home/utils/image.py", line 10, in convert_svg_to_png_bytes
    with open(svg_file_path, 'r', encoding='utf-8') as f:

Exception Type: FileNotFoundError at /en/sections/emergency-information/
Exception Value: [Errno 2] No such file or directory: '/data/idems/iogt/iogt<home.mixins.TitleIconMixin.get_icon_url.<locals>.Icon object at 0x7fcfc05de4f0>'
```

**Comment by geoo89** [comment]
Error during MZ migration:

```
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/core/management/__init__.py", line 395, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/core/management/base.py", line 330, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/core/management/base.py", line 371, in execute
    output = self.handle(*args, **options)
  File "/data/idems/iogt/iogt/iogt_content_migration/management/commands/load_v1_db.py", line 102, in handle
    self.migrate(root)
  File "/data/idems/iogt/iogt/iogt_content_migration/management/commands/load_v1_db.py", line 191, in migrate
    self.migrate_social_media_links()
  File "/data/idems/iogt/iogt/iogt_content_migration/management/commands/load_v1_db.py", line 1738, in migrate_social_media_links
    page_link_page = models.PageLinkPage(**page_link_page_data)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/wagtail/core/models.py", line 882, in __init__
    super().__init__(*args, **kwargs)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/modelcluster/models.py", line 175, in __init__
    super(ClusterableModel, self).__init__(*args, **kwargs)
  File "/data/idems/iogt/iogt_venv38/lib/python3.8/site-packages/django/db/models/base.py", line 501, in __init__
    raise TypeError("%s() got an unexpected keyword argument '%s'" % (cls.__name__, kwarg))
TypeError: PageLinkPage() got an unexpected keyword argument 'image_id'
```

**Comment by sheralim012** [comment]
@geoo89 both issues have resolved, check now

**Comment by geoo89** [comment]
All cypress tests fail now: https://github.com/sheralim012/iogt/tree/feat/external-page-link-page

On the develop branch, only the chatbot related test fails.

**Pull request merged**